src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -49,19 +49,33 @@ export class DeviceSource<T extends DeviceType> {
 export class DeviceSourceManager implements IDisposable {
     // Public Members
     /**
-     * Observable to be triggered when before a device is connected
+     * Observable to be triggered when before a device is connected, any new observers added will be triggered against already connected devices
      */
-    public readonly onBeforeDeviceConnectedObservable = new Observable<{ deviceType: DeviceType, deviceSlot: number }>();
+    public readonly onBeforeDeviceConnectedObservable = new Observable<{ deviceType: DeviceType, deviceSlot: number }>((observer) => {

```

Comment:
Could be extracted as a type or interface for ease of use ?

Reply:
How about returning DeviceSource?

Reply:
Changing this to DeviceSource